### PR TITLE
MDN CSS: HTML Escape all <input> elements.

### LINKED ISSUE #785: MDN CSS: <input> rendering
Search for [::-moz-range-track](https://duckduckgo.com/180/%3A%3A-moz-range-track)

The MDN page has the following text:
> The ::-moz-range-track CSS pseudo-element represents the track, that is the groove in which the indicator of an \<input\> of type="range" slides.

DDG is displaying:
screenshot from 2017-02-14 13-37-21

I wonder, how does DDG/zero-clicks handle XSS? Would this be at all possible to exploit?

------
IA Page:  http://duck.co/ia/view/mdn_css 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @hchienjo

### PR DESCRIPTION
**Pull request by hchienjo**
## Description of new Instant Answer, or changes

HTML Escape all <input> elements thus preventing their rendering.

## Related Issues and Discussions

Fixes #785 

## People to notify

@moollaza @pjhampton 

## Testing & Review
To be completed by Language Leader (or DDG Staff) when reviewing Pull Request

**Pull Request**
- [x] Title follows correct format (Specifies Instant Answer + Purpose)
- [x] Description contains a valid Instant Answer Page Link (e.g. https://duck.co/ia/view/my_ia)

**Instant Answer Page** (for new Instant Answers)
- [ ] Instant Answer page is correctly filled out and contains:
    - One topic for the Search Space Language (Java, Python, Scala, Ruby, etc.)
    - One topic from: Reference, Help, Libraries, Tools
        - Documentation Fatheads are considered "Reference"
    - Description, Data source, and 2+ example queries
    - Perl Module (e.g. "DDG::Fathead::PerlDoc" -- we only need a name, not an actual file)
    - Source Name (for "More at <source_name>" link)
    - Source Domain (must contain http:// or https:// -- can be the same as Data Source)
    - Source Info (used as Subtitle for each Article -- usually matches the IA Name)
    - 'Skip Abstract' is checked off
    - Source ID (ping @moollaza to assign one, once Fathead is ready for Beta deploy)

**Code**
- [x] Uniformly indented, well commented
- [x] Fetch.sh and Parse.sh run without errors
- [x] Output contains no blank lines, or multi-line entries
- [ ] Fathead Tests are passing (run `$ duckpan test <fathead_id>`)
    - Tester should report any failures

**Pull Request Review Guidelines**: https://docs.duckduckhack.com/programming-mission/pr-review.html

---

Instant Answer Page: https://duck.co/ia/view/mdn_css

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by hchienjo** [comment]
Looks like #722 is also fixed. 
screenshot_2017-02-14_17-24-11

**Comment by pjhampton** [comment]
Thanks for addressing this so fast @hchienjo!

**Comment by hchienjo** [comment]
@pjhampton Weird thing is, it works on codio, but doesn't on the deployed beta.

**Comment by pjhampton** [comment]
Hmm, I'm wondering if it actually redeployed.

File: lib/fathead/mdn_css/parse.pl [hunk 1]
```diff
@@ -297,6 +298,7 @@ sub create_abstract {
     my ( $description, $code, $initial_value ) = @_;
     if ($description) {
         $description = trim($description);
+        $description = escape_html($description) if $description =~ /</;
```
Comment (moollaza) [inline lib/fathead/mdn_css/parse.pl]: @hchienjo this approach won't work. Internally we first decode all the HTML, before re-encoding the `<pre><code>` blocks.

You'll need to wrap any html elements that are meant to be escaped in `<code>` so the browser doesn't render it

Comment (hchienjo) [inline lib/fathead/mdn_css/parse.pl]: Alright. Thanks for pointing this out. I'll work on a fix.

**Comment by hchienjo** [comment]
@moollaza I have added wrapping of `<input>` as well as `<progress>` because of https://duckduckgo.com/?q=css+%3A%3A-moz-progress-bar&t=canonical&ia=about. What do you think?

**Comment by hchienjo** [comment]
Oh no! The problem still persists on the newly deployed beta.

**Comment by hchienjo** [comment]
@moollaza What if we 'deaden' the input elements? Since they are in the description, we can make `<input>` become `input` and so on for all the elements that are causing problems.

**Comment by hchienjo** [comment]
#722 Seems resolved too.

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Pull request merged**